noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr

```diff
@@ -30,12 +31,13 @@ contract Crowdfunding {
         // Crowdfunding campaign operator
         operator: SharedImmutable<AztecAddress>,
         // End of the crowdfunding campaign after which no more donations are accepted
-        deadline: PublicImmutable<u64>,
+        deadline: SharedImmutable<u64>,
```

Comment:
Replaced PublicImmutable with SharedImmutable so that I am able to load the value in private and then pass it to the router contract.

Reply:
if we keep router to be public contract only, then we can keep this public and save on a kernel iteration + hashing cost for a shared mutable